rasterio/_warp.pyx

```diff
@@ -487,3 +491,60 @@ def _reproject(
         if dtypes.is_ndarray(source):
             if hdsin != NULL:
                 _gdal.GDALClose(hdsin)
+
+
+def _calculate_default_transform(
+        src_crs, dst_crs, width, height, left, bottom, right, top,
+        resolution=None, densify_pts=21):
```

Comment:
Do we need `densify_pts` and `resolution` here? Should we remove these kwarg since the suggested warp output has it's own mechanism for adjusting bounds/res?